Toolbox/arcsdm/areafrequency.py

```diff
@@ -39,369 +38,257 @@
-    # gp.AddMessage("got to here....11")
-    # gp.AddWarning('Assume cell size units is meters!')
-    factor = (float(gp.CellSize)**2) / 1000000 / UnitArea
-    # gp.AddMessage(str(factor))
-    # gp.AddMessage("Input_raster path=%s"%os.path.basename(Input_raster))
-
-# Search raster must be a raster layer
-    # gp.addmessage("Value type: %s"%valuetypes[valuetype])
+    # Calculate the area factor
+    factor = (float(arcpy.env.cellSize) ** 2) / 1000000 / UnitArea
+    
+    # Insert rows into the output table
     rasrows = flt_ras.FloatRasterSearchcursor()
-    # gp.addmessage('Opened Float Raster Searchcursor...')
-
-# Insert some field values in output table
-    # Open insert cursor for output
-    tblrows = gp.InsertCursor(Output_Table)
-    gp.AddMessage('factor: %s'%factor)
-    # Create Output records
+    tblrows = arcpy.InsertCursor(Output_Table)
+    arcpy.AddMessage('factor: %s' % factor)
     for rasrow in rasrows:
-        tblrow = tblrows.NewRow()
+        tblrow = tblrows.newRow()
         tblrow.RASTERVALU = rasrow.Value
-        # gp.addwarning(str(rasrow.Value))
         tblrow.Area_sqkm = rasrow.Count * factor
-        tblrows.InsertRow(tblrow)
-
-    del tblrow,tblrows
-    # gp.AddMessage("No. records in output table %s: %i"%(Output_Table,gp.GetCount_management(Output_Table)))
-
-    # Get total sites from stats table
+        tblrows.insertRow(tblrow)
+    
+    del tblrow, tblrows
+    
+    # Calculate the total number of sites
     totalsites = sum(stats_dict.values())
-    gp.AddMessage('totalsites: %s'%totalsites)
-    # Variables for more stuff
+    arcpy.AddMessage('totalsites: %s' % totalsites)
+    
+    # Initialize variables for cumulative calculations
     totalarea = 0.0
     cumArea = 0
     effarea = []
     nSites = []
     
-    # Update Frequency field and get two summations and create two lists
-    # gp.addmessage('statvals: '+str(stats_dict.keys()))
-    # gp.AddMessage("Calculating Frequency field...")
-    tblrows = gp.UpdateCursor(Output_Table)
-    tblrow = tblrows.Next()
+    # Update the output table with frequency values
+    tblrows = arcpy.UpdateCursor(Output_Table)
+    tblrow = tblrows.next()
     stats_found = 0
     while tblrow:
         tblval = tblrow.RASTERVALU
         area = tblrow.Area_sqkm
         totalarea += area
-        # tblval is less precision than rasval
         rasval = flt_ras[tblval]
         if rasval in stats_dict:
             frequency = stats_dict[rasval]
-            # gp.AddMessage("Found tblval = %s; frequency = %s"%(tblval,frequency))
             tblrow.Frequency = frequency
-            tblrows.UpdateRow(tblrow)
+            tblrows.updateRow(tblrow)
             effarea.append(area)
             nSites.append(frequency)
             stats_found += 1
-            # gp.AddMessage("Debug: Stats_found =  %s"%stats_found);
         
-        tblrow = tblrows.Next()
+        tblrow = tblrows.next()
     
-    del tblrow,tblrows
```

Comment:
I would replace this part with 
`    totalarea = 0.0
    effarea = []
    nSites = []

    with arcpy.da.UpdateCursor(Output_Table, ["RASTERVALU", "Area_sqkm", "Frequency"]) as cursor:
        for row in cursor:
            tblval = row[0]
            area = row[1]
            totalarea += area
            rasval = flt_ras[tblval]
            if rasval in stats_dict:
                frequency = stats_dict[rasval]
                row[2] = frequency
                cursor.updateRow(row)
                effarea.append(area)
                nSites.append(frequency)
                stats_found += 1`

Reply:
I have edited the code and commited new code.